hs_core/models.py

```diff
@@ -2112,7 +2112,12 @@ def delete(self, using=None):
                 # so no need for fl.logical_file.delete() and deleting of metadata file
                 # object deletes (cascade delete) all the contained GenericRelated metadata
                 # elements
-                fl.logical_file.metadata.delete()
```

Comment:
We're setting us up for a lot of bugs if every time we delete a file, we also have to check for and delete the logical_file, and also the logical file metadata and also check if the logical_file is a model instance and set the flag to dirty.

All of this should happen cascading down from deleting the resourcefile where the logical file is signaled to be delete itself, then the logical file metadata is signaled to delete itself and do any implementation specific tasks when delete occurs.  Classes should know how to delete themselves and tie into the delete signal of their parent.

Reply:
Fixed  - resource file now has the ability to delete the associated logical file